### PR TITLE
[WIN32SS:NTUSER] Level up Snapping to Windows 10

### PR DESCRIPTION
**Pull request by NikkiNext**
I made a window Snap like in Windows 10. Snap works in the upper left corner. So far, it only works with hockey. Hotkey: Win+F2

I checked on the VM. Snapping works

### CODE REVIEW

File: win32ss/user/ntuser/defwnd.c [hunk 1]
```diff
@@ -414,7 +414,7 @@ UserPaintCaption(PWND pWnd, INT Flags)
          * RealUserDrawCaption in order to draw the classic caption when themes
          * are disabled but the themes service is enabled
          */
-         TRACE("UDCB Flags %08x\n", Flags);
+         TRACE("UDCB Flags %08x\n");
```
Comment (JoachimHenze) [inline win32ss/user/ntuser/defwnd.c:417]: This change is wrong, and makes me think that you created your PR from an outdated state.

File: win32ss/user/ntuser/defwnd.c [hunk 2]
```diff
@@ -826,6 +826,7 @@ IntDefWindowProc(
                        co_IntSendMessage(hwndTop, WM_SYSCOMMAND, SC_MINIMIZE, lParam);
                    }
                }
+
```
Comment (JoachimHenze) [inline win32ss/user/ntuser/defwnd.c:829]: This change makes no sense.

**Review by JoachimHenze — CHANGES_REQUESTED** [review]

File: win32ss/user/ntuser/defwnd.c [hunk 3]
```diff
@@ -825,7 +825,7 @@ IntDefWindowProc(
                    {
                        co_IntSendMessage(hwndTop, WM_SYSCOMMAND, SC_MINIMIZE, lParam);
                    }
-               }
+               }
```
Comment (SergeGautherie) [inline win32ss/user/ntuser/defwnd.c:828]: No.

File: win32ss/user/ntuser/defwnd.c [hunk 4]
```diff
@@ -863,10 +863,15 @@ IntDefWindowProc(
                   {
                      snapRect.right = (snapRect.left + snapRect.right) / 2;
                   }
-                  else // VK_RIGHT
+                  if (wParam == VK_RIGHT) // VK_RIGHT
```
Comment (SergeGautherie) [inline win32ss/user/ntuser/defwnd.c:866]:
```suggestion
                  else if (wParam == VK_RIGHT) // VK_RIGHT
```

File: win32ss/user/ntuser/defwnd.c [hunk 5]
```diff
@@ -863,10 +863,15 @@ IntDefWindowProc(
...
                   {
                      snapRect.left = (snapRect.left + snapRect.right) / 2;
                   }
+				  if (wParam == VK_F2)
```
Comment (SergeGautherie) [inline win32ss/user/ntuser/defwnd.c:870]:
```suggestion
				  else if (wParam == VK_F2)
```

**Comment by SergeGautherie** [comment]
Use spaces, not tabulations.

File: win32ss/user/ntuser/defwnd.c (hunk 5 above)
Comment (learn-more) [reply win32ss/user/ntuser/defwnd.c:870]: because any key is the same as VK_F2?

**Review by learn-more — CHANGES_REQUESTED** [review]
Hello,
Thanks for your interest in ReactOS!
We ask our contributors [to use their full name when committing](https://github.com/reactos/reactos/blob/master/CONTRIBUTING.md#rules-and-recommendations).
[Please amend your commit with your full name](https://stackoverflow.com/a/3042512), and update this PR.

To change this for future PR's, you can update it for globally (for the system):
```
git config --global user.name "Your Name"
git config --global user.email you@example.com
```

Or just for the current repository:
```
git config user.name "Your Name"
git config user.email you@example.com
```

Regards,

File: win32ss/user/ntuser/defwnd.c [hunk 6]
```diff
@@ -863,10 +864,15 @@ IntDefWindowProc(
...
+                  {
+                     snapRect.right = (snapRect.left + snapRect.right) / 2;
+					 snapRect.bottom = (snapRect.top + snapRect.bottom) / 2;
```
Comment (HBelusca) [inline win32ss/user/ntuser/defwnd.c:874]:
```suggestion
                     snapRect.bottom = (snapRect.top + snapRect.bottom) / 2;
```
(fixed tabs vs. spaces)

**Comment by HBelusca** [comment]
I don't understand why the hotkey "Win+F2". On my windows 10 machine this doesn't do anything. If I want to use hotkeys to snap on the top-left I press successively "Win+LeftArrow" (twice) then "Win+UpArrow".

File: win32ss/user/ntuser/input.c [hunk 7]
```diff
@@ -203,6 +203,7 @@ RawInputThreadMain(VOID)
                 UserRegisterHotKey(PWND_BOTTOM, IDHK_SNAP_RIGHT, MOD_WIN, VK_RIGHT);
                 UserRegisterHotKey(PWND_BOTTOM, IDHK_SNAP_UP, MOD_WIN, VK_UP);
                 UserRegisterHotKey(PWND_BOTTOM, IDHK_SNAP_DOWN, MOD_WIN, VK_DOWN);
+				UserRegisterHotKey(PWND_BOTTOM, IDHK_SNAP_LEFTUP, MOD_WIN, VK_F2);
```
Comment (HBelusca) [inline win32ss/user/ntuser/input.c:206]:
```suggestion
               UserRegisterHotKey(PWND_BOTTOM, IDHK_SNAP_LEFTUP, MOD_WIN, VK_F2);
```

File: win32ss/user/ntuser/input.c [hunk 8]
```diff
@@ -203,6 +203,7 @@ RawInputThreadMain(VOID)
                 UserRegisterHotKey(PWND_BOTTOM, IDHK_SNAP_RIGHT, MOD_WIN, VK_RIGHT);
                 UserRegisterHotKey(PWND_BOTTOM, IDHK_SNAP_UP, MOD_WIN, VK_UP);
                 UserRegisterHotKey(PWND_BOTTOM, IDHK_SNAP_DOWN, MOD_WIN, VK_DOWN);
+               UserRegisterHotKey(PWND_BOTTOM, IDHK_SNAP_LEFTUP, MOD_WIN, VK_F2);
```
Comment (SergeGautherie) [inline win32ss/user/ntuser/input.c:206]: Misaligned.

**Review by binarymaster — CHANGES_REQUESTED** [review]
Did you added unrelated changes to your PR accidentally? Please remove them. Also this PR won't be merged until you address request from Mark.

**Comment by NikkiNext** [comment]
@learn-more @binarymaster I'm sorry for the confusion. But can you check if my email and name are displayed on Github patches?

**Comment by NikkiNext** [comment]
I temporarily closed it in order not to mix everything into a mess

**Pull request closed without merging**